static/app/views/settings/organizationTeams/teamMembersRow.tsx

```diff
@@ -54,29 +66,39 @@ const TeamRoleSelect = (props: {
   member: TeamMember;
   organization: Organization;
   updateMemberRole: (member: TeamMember, newRole: string) => void;
+  orgRoleFromTeam?: string;
 }) => {
-  const {hasWriteAccess, organization, member, updateMemberRole} = props;
+  const {hasWriteAccess, organization, orgRoleFromTeam, member, updateMemberRole} = props;
   const {orgRoleList, teamRoleList, features} = organization;
   if (!features.includes('team-roles')) {
     return null;
   }
 
-  const {orgRole: orgRoleId} = member;
-  const orgRole = orgRoleList.find(r => r.id === orgRoleId);
+  // Determine the team-role, including if the current team has an org role
+  // and if adding the user to the current team changes their minimum team-role
+  const {orgRolesFromTeams} = member;
+  const memberOrgRoles = [member.orgRole];
+  if (orgRolesFromTeams) {
+    memberOrgRoles.push(orgRolesFromTeams[0].role.id);
+  }
+  if (orgRoleFromTeam) {
+    memberOrgRoles.push(orgRoleFromTeam);
+  }
+  const topOrgRole = getTopOrgRole(memberOrgRoles, orgRoleList);
```

Comment:
This chunk is unnecessarily confusing. I don't know what's the difference between `orgRolesFromTeams` and `orgRoleFromTeam`.

I rather that you pass in `team` entirely above, like this
```
<TeamRoleSelect
  ...
  team={team}
  member={member}
/>
```

Then for this chunk

```
const possibleOrgRoles = [member.orgRole];
if (member.orgRolesFromTeams) {
  possibleOrgRoles.push(member.orgRolesFromTeams[0].role.id);
}
if (team.orgRole) {
  possibleOrgRoles.push(team.orgRole);
}
const effectiveOrgRole = getTopOrgRole(possibleOrgRoles, orgRoleList);
```

Might be better to rename `getTopOrgRole` to `getEffectiveOrgRole` or `getActualOrgRole`

Reply:
we use top dog so i feel like top org role is not too much of a stretch